vms/evm/predicate/README.md

```diff
@@ -0,0 +1,141 @@
+0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00,
+// Byte array results (value in TxPredicateResults map)
+// Length of bytes result
+0x00, 0x00, 0x00, 0x03,
+// bytes
+0x01, 0x02, 0x03
+```
+
+##### Predicate Map with Two Transaction Results
+
+```
+// codecID
+0x00, 0x00,
+// Results length
+0x00, 0x00, 0x00, 0x02,
+// txHash (key in results map)
+0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+// TxPredicateResults (value in results map)
+// TxPredicateResults length
+0x00, 0x00, 0x00, 0x01,
+// precompile address (key in TxPredicateResults map)
+0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00,
+// Byte array results (value in TxPredicateResults map)
+// Length of bytes result
+0x00, 0x00, 0x00, 0x03,
+// bytes
+0x01, 0x02, 0x03
+// txHash2 (key in results map)
+0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+// TxPredicateResults (value in results map)
+// TxPredicateResults length
+0x00, 0x00, 0x00, 0x01,
+// precompile address (key in TxPredicateResults map)
+0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00,
+// Byte array results (value in TxPredicateResults map)
+// Length of bytes result
+0x00, 0x00, 0x00, 0x03,
+// bytes
+0x01, 0x02, 0x03
+```
+
+#### Maximum Size
+
+Results has a maximum size of 1MB enforced by the codec. The actual size depends on how much data the Precompile predicates may put into the results, the gas cost they charge, and the block gas limit.
+
+The Results maximum size should comfortably exceed the maximum value that could happen in practice, so that a correct block builder will not attempt to build a block and fail to marshal the predicate results using the codec.
+
+We make this easy to reason about by assigning a minimum gas cost to the `PredicateGas` function of precompiles. In the case of Warp, the minimum gas cost is set to 200k gas, which can lead to at most 32 additional bytes being included in Results.
+
+The additional bytes come from the transaction hash (32 bytes), length of tx predicate results (4 bytes), the precompile address (20 bytes), length of the bytes result (4 bytes), and the additional byte in the results bitset (1 byte). This results in 200k gas contributing a maximum of 61 additional bytes to Result.
+
+For a block with a maximum gas limit of 100M, the block can include up to 500 validated predicates based contributing to the size of Result. At 61 bytes / validated predicate, this yields ~30KB, which is well short of the 1MB cap.
```

Comment:
My understanding is that the audience for this doc is people trying to integrate with our data format and need to know how to parse this data. It feels like this doc is being ostentatious with its language when we could just say that the max size is 1 MiB.

Reply:
Would it make sense to delete this section and move the content then? I think the maximum size of 1MIB is relevant to the audience and can be moved to the beginning of the serialization header, and the historical context for why this magic number was chosen can be left as a code comment over the the variable. Interested readers can still discover the information - but this information isn't really relevant for the user who just needs to know how to unmarshal our data.